spec/property_collection_spec.rb

```diff
@@ -36,4 +36,14 @@
       expect(base_class.new).to have_smart_property(:title)
     end.not_to raise_error(SystemStackError)
   end
+
+  specify "PropertyCollection has O(1) #count" do
+    property_collection = Class.new(SmartProperties::PropertyCollection) do
+      def collection_with_parent_collection
+        mock('collection').expects(:count).once
```

Comment:
Maybe we should add an additional return value and below assert for that:

```ruby
mock('collection').expects(:count).once.returns(42)
```

and

```
expect(property_collection.new.count).to eq(42)
```